invokeai/app/services/bulk_download/bulk_download_default.py

```diff
@@ -0,0 +1,176 @@
+from pathlib import Path
+from tempfile import TemporaryDirectory
+from typing import Optional, Union
+from zipfile import ZipFile
+
+from invokeai.app.services.board_records.board_records_common import BoardRecordNotFoundException
+from invokeai.app.services.bulk_download.bulk_download_common import (
+    DEFAULT_BULK_DOWNLOAD_ID,
+    BulkDownloadException,
+    BulkDownloadTargetException,
+)
+from invokeai.app.services.events.events_base import EventServiceBase
+from invokeai.app.services.image_records.image_records_common import ImageRecordNotFoundException
+from invokeai.app.services.images.images_common import ImageDTO
+from invokeai.app.services.invoker import Invoker
+from invokeai.app.util.misc import uuid_string
+
+from .bulk_download_base import BulkDownloadBase
+
+
+class BulkDownloadService(BulkDownloadBase):
+    __output_folder: Path
+    __temp_directory: TemporaryDirectory
+    __bulk_downloads_folder: Path
+    __event_bus: EventServiceBase
+    __invoker: Invoker
+
+    def start(self, invoker: Invoker) -> None:
+        self.__invoker = invoker
+        self.__event_bus = invoker.services.events
+
+    def __init__(
+        self,
+        output_folder: Union[str, Path],
+    ):
+        """
+        Initialize the downloader object.
+        """
+        self.__output_folder: Path = output_folder if isinstance(output_folder, Path) else Path(output_folder)
+        self.__temp_directory = TemporaryDirectory(dir=self.__output_folder)
+        self.__bulk_downloads_folder = Path(self.__temp_directory.name) / "bulk_downloads"
+        self.__bulk_downloads_folder.mkdir(parents=True, exist_ok=True)
+
+    def handler(self, image_names: list[str], board_id: Optional[str], bulk_download_item_id: Optional[str]) -> None:
```

Comment:
It may make sense to split `handler` up for each type of bulk download. No need to worry about null checks.

That would also allow `get_clean_board_name` to be an implementation detail of the bulk_download service, because it's not needed in the router, reducing the scope of the base class a bit.

I also think the download item id should be a UUID to avoid the need to worry about collisions. Right now we overwrite existing files, but I wonder what happens if two requests are made simultaneously. Using a simple UUID shifts the responsibility of naming the zip file to the handler, so the router doesn't need to worry about figuring out the board name.

Then, the filename could be `f"{board_name_clean}_{uuid}.zip"` for board downloads, or just `f"{uuid}.zip"`

Reply:
Yeah originally it was just UUID, but I had thought that perhaps it would be nice to see the board name as the zip file. I can revert to uuid.

Reply:
In the current implementation the `bulk_download_item_id` is either the uuid or the board name and because that it is returned by the API `get_clean_board_name` will continue to be an implementation detail of `bulk_download_base`. 

It is certainly not ideal and kinda blurs the lines of separation of duties. Maybe the better choice would be to create a method called `generate_bulk_download_item_id` in the bulk_download_base and give the clean file name responsibility to the service rather than the path.